samples/extensions/AWSSQS/trigger/sqsreceivemessage/trigger.go

```diff
@@ -0,0 +1,216 @@
+        }
+        if handlerSetting.QueueURL == "" {
+            return errors.New("Empty Queue URL")
+        }
+        sqsHandler := &SQSHandler{}
+        sqsHandler.settings = handlerSetting
+        sqsHandler.handler = handler
+        sqsHandler.shutdown = make(chan bool)
+        sqsHandler.sqssvc = t.sqssvc
+        sqsHandler.triggerName = t.name
+        t.SQSHandlers[handlerSetting.QueueURL] = sqsHandler
+    }
+    return err
+}
+
+func (t *Trigger) Start() error {
+    for _, handler := range t.SQSHandlers {
+        go handler.start()
+    }
+    return nil
+}
+
+func (t *Trigger) Stop() error {
+    for _, handler := range t.SQSHandlers {
+        handler.shutdown <- true
+    }
+    return nil
+}
+
+
+func (h *SQSHandler) start() {
+    for {
+        select {
+        case <-h.shutdown:
+            triggerLog.Debugf("Stopping receiver for Queue [%s] for trigger [%s]", h.settings.QueueURL, h.triggerName)
+            return
+        default:
+            sqsSvc := h.sqssvc
+            receiveMessageInput := &sqs.ReceiveMessageInput{}
+            receiveMessageInput.QueueUrl = aws.String(h.settings.QueueURL)
+
+            attrsNames := h.settings.AttributeNames
+            if attrsNames != nil && len(attrsNames) > 0 {
+                //Add attribute names
+                attrs := make([]*string, len(attrsNames))
+                for i, v := range attrsNames {
+                    attrInfo, _ := coerce.ToObject(v)
+                    if attrInfo != nil && attrInfo["Name"] != nil {
+                        attrs[i] = aws.String(attrInfo["Name"].(string))
+                    }
+                }
+                receiveMessageInput.AttributeNames = attrs
+            }
+
+            attrsNames = h.settings.MessageAttributeNames
+            if attrsNames != nil && len(attrsNames) > 0 {
+                attrs := make([]*string, len(attrsNames))
+                for i, v := range attrsNames {
+                    attrInfo, _ := coerce.ToObject(v)
+                    if attrInfo != nil && attrInfo["Name"] != nil {
+                        attrs[i] = aws.String(attrInfo["Name"].(string))
+                    }
+                }
+                receiveMessageInput.MessageAttributeNames = attrs
+            }
+
+
+            maxNumberOfMessages := h.settings.MaxNumberOfMessages
+            if maxNumberOfMessages != 0 {
+                receiveMessageInput.MaxNumberOfMessages = aws.Int64(int64(maxNumberOfMessages))
+            }
+
+            visibilityTimeout := h.settings.VisibilityTimeout
+            if visibilityTimeout != 0 {
+                receiveMessageInput.VisibilityTimeout = aws.Int64(int64(visibilityTimeout))
+            }
+
+            waitTimeSeconds := h.settings.WaitTimeSeconds
+            if waitTimeSeconds != 0 {
+                receiveMessageInput.WaitTimeSeconds = aws.Int64(int64(waitTimeSeconds))
+            }
+
+            response, err := sqsSvc.ReceiveMessage(receiveMessageInput)
```

Comment:
Let's create input message during initialization as it is same throughout.

Reply:
ok

Reply:
Have done the changes, tested it working fine.